src/storage/page_buffer.c

```diff
@@ -2074,6 +2092,17 @@ pgbuf_unfix (THREAD_ENTRY * thread_p, PAGE_PTR pgptr)
 	{
 	  return;
 	}
+
+      holder = pgbuf_get_holder (thread_p, pgptr);
+
+      assert_release (holder != NULL);
+
+      watcher = holder->last_watcher;
+      while (watcher != NULL)
+	{
+	  assert (watcher->magic == PGBUF_WATCHER_MAGIC_NUMBER);
+	  watcher = watcher->prev;
+	}
```

Comment:
I think it is better to separate the new block from `pgbuf_get_check_page_validation`.
```c
  if (pgbuf_get_check_page_validation (thread_p, PGBUF_DEBUG_PAGE_VALIDATION_FREE))
    {
      ...
    }

holder = pgbuf_get_holder (thread_p, pgptr);

assert (holder != NULL);

watcher = holder->last_watcher;
while (watcher != NULL)
  {
    assert (watcher->magic == PGBUF_WATCHER_MAGIC_NUMBER);
    watcher = watcher->prev;
  }
```